iib/web/models.py

```diff
@@ -752,11 +753,16 @@ def _from_json(
         Architecture.validate_architecture_json(add_arches)
 
         # Validate binary_image is correctly provided
-        binary_image = request_kwargs.pop('binary_image')
-        if not isinstance(binary_image, str):
+        binary_image = request_kwargs.pop('binary_image', None)
+        if binary_image and not isinstance(binary_image, str):
```

Comment:
This will catch falsy values of the wrong type
```suggestion
        if binary_image is not None and not isinstance(binary_image, str):
```